### PR TITLE
清理 Core 模块

### PR DESCRIPTION
**Pull request by pandamicro**
> Makes sure these boxes are checked before submitting your PR - thank you!
> - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
> - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
> - [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
> - [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.

Re: cocos-creator/fireball#1785
Re: cocos-creator/fireball#481

Changes proposed in this pull request:
- 模块化 event-manager, renderer
- 删除 Base64Images
- 删除 CCProgressTimer
- 删除 SpriteFrameAnimation
- 清理 CCSGNode，CCSGEditBox，CCEventListener，CCEventManager，CCSGLabel，CCLayer，CCEGLView，CCSGScene，CCScale9Sprite，CCSGSprite，CCSpriteBatchNode，CCSpriteFrame，CCVertex，CCTexture2D，CCGrid 中的常量，无用代码，局部变量。
- 清理并模块化 cc.math

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/base-nodes/CCSGNode.js [hunk 1]
```diff
@@ -1233,7 +1226,7 @@ _ccsg.Node = cc.Class({
      * @see _ccsg.Node#removeChildByTag
      */
     removeChildByTag: function (tag, cleanup) {
-        if (tag === cc.NODE_TAG_INVALID)
+        if (tag === cc.Macro.NODE_TAG_INVALID)
```
Comment (jareguo) [inline cocos2d/core/base-nodes/CCSGNode.js]: 我其实不太能理解，为什么 Macro 就要大写，math, js 就要小写

Comment (pandamicro) [inline cocos2d/core/base-nodes/CCSGNode.js]: 呃。。。好像确实应该小写，我没多想

File: gulp/tasks/build-cocos2d.js [hunk 2]
```diff
@@ -126,11 +95,6 @@ var srcs = [
 
     './cocos2d/core/CCActionManager.js',
 
-    './cocos2d/progress-timer/CCProgressTimer.js',
-    './cocos2d/progress-timer/CCProgressTimerCanvasRenderCmd.js',
-    './cocos2d/progress-timer/CCProgressTimerWebGLRenderCmd.js',
-    './cocos2d/progress-timer/CCActionProgressTimer.js',
-
```
Comment (jareguo) [inline gulp/tasks/build-cocos2d.js:133]: 这里面的所有文件，之前在 core-level 都是不加载的

Comment (pandamicro) [inline gulp/tasks/build-cocos2d.js:133]: 收到

**Pull request merged**